### PR TITLE
Add nvfp4_omlp_only config and simplify the config.py

### PR DESCRIPTION
**Pull request by cjluo-nv**
### What does this PR do?

Type of change: ? new feature

1) Add vfp4_omlp_only config == nvfp4_mlp_only + o_proj quant
2) Add block sparse MOE to mlp only config
3) Simplfiy config.py
4) Update readme in llm_ptq mention these two configs for better accuracy.

### Usage

huggingface_script.sh ... --quant nvfp4_omlp_only

### Before your PR is "*Ready for review*"

Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed (`git commit -s -S`).

Make sure you read and follow the [Security Best Practices](https://github.com/NVIDIA/Model-Optimizer/blob/main/SECURITY.md#security-coding-practices-for-contributors) (e.g. avoiding hardcoded `trust_remote_code=True`, using `torch.load(..., weights_only=True)`, avoiding `pickle`, etc.).

- Is this change backward compatible?: ✅ / ❌ / N/A 
- If you copied code from any other source, did you follow IP policy in [CONTRIBUTING.md](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md#-copying-code-from-other-sources)?: ✅ / ❌ / N/A 
- Did you write any new necessary tests?: ✅ / ❌ / N/A 
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: ✅ / ❌ / N/A 

## Summary by CodeRabbit

* **New Features**
  * Added nvfp4_omlp_only quantization format for NVFP4, enabling selective quantization of MLP and output projection layers while preserving attention QKV projection accuracy.

* **Changed**
  * pass_through_bwd now defaults to True; set to False if using STE with zeroed outlier gradients for better QAT accuracy.

* **Documentation**
  * Updated post-training quantization guidance with NVFP4-specific configuration recommendations and usage examples.

### CODE REVIEW

**Review by jenchen13 — APPROVED** [review]

File: modelopt/torch/quantization/config.py [hunk 1]
```diff
@@ -713,29 +597,38 @@
...
-            "enable": True,
-            "pass_through_bwd": True,
-        },
-        **_default_disabled_quantizer_cfg,
+        "*o_proj*weight_quantizer": _nvfp4_quantizer,
+        "*o_proj*input_quantizer": _nvfp4_quantizer,
+        **_nvfp4_mlp_only_quant_cfg,
     },
     "algorithm": "max",
 }
```
[the remark this answers is not in the document]
Comment (jenchen13) [reply modelopt/torch/quantization/config.py:659]: ^ update `choices` below with new configs

File: modelopt/torch/quantization/config.py [hunk 2]
```diff
@@ -1074,14 +967,14 @@ def validate_calibrator(cls, v, info: ValidationInfo):
...
         title="If set to true, fake quantization will be a pass through for gradient computation.",
         description="""
         Gradient computation where fake quantization is pass through is called
         'Straight-Through Estimator (STE)'. STE does not require saving of the input tensor for
         performing backward pass and hence consumes less memory.
 
-        If set to False, we will use STE with zeroed outlier gradients. This setting could
+        If set to False, we will use STE with zeroed outlier gradients. This setting may
         yield better QAT accuracy depending on the quantization format. However, this setting
         requires saving of the input tensor for computing gradients which uses more memory.
```
[the remark this answers is not in the document]
Comment (jenchen13) [reply modelopt/torch/quantization/config.py:1006]: should this `pass_through_bwd` change be documented in the readme or changelog as well?

File: modelopt/torch/quantization/config.py [hunk 3]
```diff
@@ -392,28 +389,22 @@
...
-        },
-        "*input_quantizer": {
-            "num_bits": (2, 1),
-            "block_sizes": {-1: 16, "type": "dynamic", "scale_bits": (4, 3)},
-            "axis": None,
-            "enable": True,
-        },
+        "*weight_quantizer": _nvfp4_quantizer,
+        "*input_quantizer": _nvfp4_quantizer,
         **_default_disabled_quantizer_cfg,
```
[the remark this answers is not in the document]
Comment (cjluo-nv) [reply modelopt/torch/quantization/config.py:425]: ACK

File: modelopt/torch/quantization/config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cjluo-nv) [reply modelopt/torch/quantization/config.py:1006]: Updated change log

File: examples/llm_ptq/example_utils.py [hunk 4]
```diff
@@ -258,18 +258,6 @@ def build_quant_cfg(
...
-        quant_cfg["quant_cfg"]["model*.*attn*q_proj*"] = {"enable": False}
-        quant_cfg["quant_cfg"]["model*.*attn*k_proj*"] = {"enable": False}
-        quant_cfg["quant_cfg"]["model*.*attn*v_proj*"] = {"enable": False}
-
-    if model_type == "deepseek":
-        # Disable MLA quantization for accuracy.
-        quant_cfg["quant_cfg"]["*self_attn.q*"] = {"enable": False}
-        quant_cfg["quant_cfg"]["*self_attn.kv*"] = {"enable": False}
-
     return quant_cfg
```
[the remark this answers is not in the document]
Comment (cjluo-nv) [reply examples/llm_ptq/example_utils.py:261]: ACK

File: modelopt/torch/quantization/config.py [hunk 5]
```diff
@@ -1074,14 +968,14 @@ def validate_calibrator(cls, v, info: ValidationInfo):
...
         title="If set to true, fake quantization will be a pass through for gradient computation.",
         description="""
         Gradient computation where fake quantization is pass through is called
         'Straight-Through Estimator (STE)'. STE does not require saving of the input tensor for
         performing backward pass and hence consumes less memory.
 
-        If set to False, we will use STE with zeroed outlier gradients. This setting could
+        If set to False, we will use STE with zeroed outlier gradients. This setting may
         yield better QAT accuracy depending on the quantization format. However, this setting
         requires saving of the input tensor for computing gradients which uses more memory.
```
[the remark this answers is not in the document]
Comment (cjluo-nv) [reply modelopt/torch/quantization/config.py:1006]: ACK

File: modelopt/torch/quantization/conversion.py [hunk 6]
```diff
@@ -211,6 +211,20 @@ def _replace_quant_module(model: nn.Module, version=None, registry=QuantModuleRe
...
+
+    When ``"default": {"enable": False}`` disables all quantizers and a specific pattern then
+    sets properties like ``num_bits`` / ``block_sizes``, the quantizer should be re-enabled
+    implicitly rather than requiring an explicit ``"enable": True`` in every config entry.
+    """
+    if isinstance(cfg, dict) and "enable" not in cfg and cfg:
+        cfg = {**cfg, "enable": True}
+    elif isinstance(cfg, list):
+        cfg = [_auto_enable_cfg(c) for c in cfg]
+    return cfg
```
Comment (realAsma) [inline modelopt/torch/quantization/conversion.py]: Can we revert removing the `"enable": True` please? It is creating unnecessary  complications and can break things.

Comment (realAsma) [reply modelopt/torch/quantization/conversion.py]: We should be striving towards making Model Optimizer more easier to read - hacks like these hidden somewhere are not good for readability and driving external contributions.

**Comment by cjluo-nv** [comment]
> Can we revert removing `"enable": True`, please? This behavior adds unwarranted complications.
> 
> If the objective of this PR is to reduce the length of config.py, that’s not the right goal.
> 
> We’re moving toward YAML-based recipes, which will address the recipe expansion issue.

Done
